app/src/main/java/com/cp/tasky/core/data/util/NetworkErrorMessage.kt

```diff
@@ -0,0 +1,7 @@
+package com.cp.tasky.core.data.util
+
+object NetworkErrorMessage {
+    const val INVALID_EMAIL_OR_PASSWORD = "Invalid email or password"
+    const val SERVER_NOT_AVAILABLE = "Server not available"
+    const val UNKNOWN_ERROR = "Unknown error"
+}
```

Comment:
What are these messages used for? Are these displayed in the UI? If yes, I would leave it up to the UI to construct these strings, since otherwise you'll lose the option to localize your app, since you need string resources for that. These messages would always be in english in that case.

For passing the network results to the UI, I love this generic result class. You can let your repositories return it depending on the status code:
```
sealed interface Result<out D, out E: Error> {
    data class Success<out D>(val data: D): Result<D, Nothing>
    data class Error<out E: com.plcoding.tasky.core.domain.util.Error>(val error: E): Result<Nothing, E>
}

inline fun <T, E: Error, R> Result<T, E>.map(map: (T) -> R): Result<R, E> {
    return when(this) {
        is Result.Error -> Result.Error(error)
        is Result.Success -> Result.Success(map(data))
    }
}

fun <T, E: Error> Result<T, E>.asEmptyDataResult(): EmptyResult<E> {
    return map {  }
}

inline fun <T, E: Error> Result<T, E>.onSuccess(action: (T) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> this
        is Result.Success -> {
            action(data)
            this
        }
    }
}
inline fun <T, E: Error> Result<T, E>.onError(action: (E) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> {
            action(error)
            this
        }
        is Result.Success -> this
    }
}

typealias EmptyResult<E> = Result<Unit, E>
```
It handles data and errors in a type-safe way. You'll need a marker interface to mark error enums:
```
interface Error
```
And then you can extend this across the codebase with different error types, e.g.:
```
sealed interface DataError: Error {
    enum class Remote: DataError {
        NOT_FOUND,
        NO_INTERNET,
        UNAUTHORIZED,
        ...
    }
    
    enum class Local: DataError {
        DISK_FULL,
        ...
    }
}
```
Works great:
```
// In ViewModel
repository.signUp(payload).onSuccess { responseData ->

}.onError { error -> 
    _state.update { it.copy(error = error) }
}
```